data/articles/progress-bar.md

```diff
@@ -0,0 +1,395 @@
+
+For brevity's sake I've put everything in the perform method, but it in real life it could be a good idea to split it out more into methods, for readability and clearer error outputs.
+
+```ruby
+module Main
+  module Workers
+    class IsbnSearch
+      include Sidekiq::Job
+      include Deps[
+                "services.get_google_isbn",
+                "redis",
+                "persistence.rom",
+                "repositories.books",
+                parser: "parsers.google.isbn"]
+
+      def perform(isbn)
+        redis.hset("isbn_search", { isbn => 1 })
+
+        output = parser.parse(json: get_google_isbn.call(isbn:))
+        
+        redis.hset("isbn_search", { isbn => 2 })
+
+        if books.by_isbn(type: 10, identifier: isbn) || books.by_isbn(type: 13, identifier: isbn)
+          redis.hset("isbn_search", { isbn => 3 })
+        else
+          rom.relations[:books].transaction do
+            author = rom.relations[:authors].changeset(:create, { name: output[:authors].join(', ') }).commit
+            new_book = rom.relations[:books].changeset(:create, output[:data]).commit
+            redis.hset("isbn_search", { isbn => 3 })
+          end
+        end
+      end
+    end
+  end
+end
+
+```
+
+Now this is a fully functional progress bar (the implementations of parser and get_google_isbn objects are not really relevant).
+
+This could be further improved and made into a far more reusable code, with redis communication being delegated to pub/sub system, that could be better coupled with places that do actually push the progress forward, rather than putting everything into a worker, that should be more of a simple delegator.
+
+Now lets see how this looks in action (please not that I am not exactly a CSS wizard, or an UI designer so the styling is not the best, but it works!):
+
+
+![Progress Bar gif](/images/articles/progress-bar/demo.gif)
+
+
+
+### Summary
+
+I hope you've enjoyed this episode, and if you want to see more content in this fashion,** Subscribe to [this YT channel](https://www.youtube.com/c/HanamiMastery)** and **follow me [on twitter](twitter.com/hanamimastery)**!  As always, all links you can find the description of the video or in the the https://hanamimastery.com.
+
+### Special Thanks!
+
+I'd like to thank [LATEST SPONSORS]. for supporting this project!
+
+Any support allows me to spend more time on creating this content, promoting great open source projects.
+
+Also, check out two of my previous videos here! Thank you all for being here, you're awesome! - and see you in the next Hanami Mastery episode!
+
+- [John Smith]()- for a great cover image
```

Comment:
Those sections we  can remove, the main person I'd thank for this episode would be you!